### PR TITLE
ser2net: init at 4.1.1 (with gensio)

### PR DESCRIPTION
**Pull request by Emantor**
###### Motivation for this change
Add the ser2net serial to network server. Also packages the library it depends on.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gensio/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, lib, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "gensio";
+  version = "1.3.3";
+
+  src = fetchurl {
+    url = "https://downloads.sourceforge.net/sourceforge/ser2net/${pname}-${version}.tar.gz";
```
Comment (misuzu) [inline pkgs/development/libraries/gensio/default.nix]: Consider using github repository: https://github.com/cminyard/ser2net

File: pkgs/servers/ser2net/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, lib, fetchurl, gensio, libyaml }:
+
+stdenv.mkDerivation rec {
+  pname = "ser2net";
+  version = "4.1.1";
+
+  src = fetchurl {
+    url = "https://downloads.sourceforge.net/sourceforge/ser2net/${pname}-${version}.tar.gz";
```
Comment (misuzu) [inline pkgs/servers/ser2net/default.nix]: Consider using github repository: https://github.com/cminyard/gensio

File: pkgs/development/libraries/gensio/default.nix (hunk 1 above)
Comment (Emantor) [reply pkgs/development/libraries/gensio/default.nix]: Done, thanks for the review.

File: pkgs/servers/ser2net/default.nix (hunk 2 above)
Comment (Emantor) [reply pkgs/servers/ser2net/default.nix]: Done, thanks for the review.

**Review by misuzu — APPROVED** [review]

**Pull request merged**